lib/cf_deployer/driver/cloud_formation_driver.rb

```diff
@@ -21,13 +21,18 @@ def update_stack template, opts
           CfDeployer::Driver::DryRun.guard "Skipping update_stack" do
             aws_stack.update opts.merge(:template => template)
           end
+
+          return !CfDeployer::Driver::DryRun.enabled?
         rescue AWS::CloudFormation::Errors::ValidationError => e
           if e.message =~ /No updates are to be performed/
             Log.info e.message
+            return false
           else
             raise
           end
         end
+
+        true
```

Comment:
>  One such case to get to that line of code would be to 
> have an exception that is not a "No updates are to be performed"

* if it's not a `AWS::CloudFormation::Errors::ValidationError`, it will just continue raising, not return True.
* if it's a `AWS::CloudFormation::Errors::ValidationError` error, again, it raises, not return True.

your rewrite looks fine.